codegen/src/main/twirl/templates/ScalaServer/Handler.scala.txt

```diff
@@ -128,12 +128,24 @@ object @{serviceName}Handler {
           .recoverWith(GrpcExceptionHandler.from(eHandler(system.classicSystem))(system, writer))
       ).getOrElse(unsupportedMediaType)
 
-      Function.unlift((req: model.HttpRequest) => req.uri.path match {
+     @for(method <- service.methods if !method.inputStreaming) {
+      lazy val @{method.name}Handler = @if(method.outputStreaming) {
+             akka.grpc.HttpApi.getSourceHandler@{if(powerApis) "PowerApi" else ""}("@{method.name}", @{service.name}.descriptor, implementation.@{method.nameSafe})
+               } else {
+             akka.grpc.HttpApi.getHttpFutureHandler@{if(powerApis) "PowerApi" else ""}("@{method.name}", @{service.name}.descriptor, implementation.@{method.nameSafe})
+               }
+     }
+
+      val httpApiHandlers = (List.empty[PartialFunction[model.HttpRequest, scala.concurrent.Future[model.HttpResponse]]]  @for(method <- service.methods if !method.inputStreaming){ ++ @{method.name}Handler})
+        .foldLeft(PartialFunction.empty[model.HttpRequest, scala.concurrent.Future[model.HttpResponse]])((acc, p) => acc.orElse(p))
```

Comment:
It's very likely that you'll want to use a different method of partial function composition (I remember that I had to do something more custom to make sure that it was well-behaved in the face of many endpoints when I did the Cloudstate version). Otherwise you'll have to do a match linearly against all possible cases for each request.

Reply:
Yes, you are right. I've changed my implementation. I can get grpc handler and use it, so what I need to do is transforming json http request to grpc request and handle it with grpc handler, then transform grpc response to json http response. Not too efficient, but it might work.